File: C/exercice2_4.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <errno.h>

int numero_fils = 0 ;

int main()
{
	pid_t pid_fils ;
	int i = 0 ;
	for (i = 1; i <= 5; ++i)
	{
		numero_fils++ ;
		pid_fils = fork() ;
		if (pid_fils == 0)
		{
			printf("je suis le fils %d\n", numero_fils) ;
			return (0) ;
		}
	}
	for (i = 1; i <= 5; ++i)
	{
		wait(NULL) ;
	}
	return (0) ;
}

File: Python/exercice2_4.py
import os
import sys
import time

numero_fils = 0

def main():
    global numero_fils
    
    for i in range(1, 6):
        numero_fils += 1
        pid_fils = os.fork()
        
        if pid_fils == 0:
            print(f"je suis le fils {numero_fils}")
            return
    
    for i in range(1, 6):
        os.wait()

if __name__ == "__main__":
    main()